File: cpp/foxford_2nd_year/9/2.cpp
#include <iostream>

using namespace std;

int extended_gcd(int a, int b, int &x, int &y)
{
    if (a == 0)
    {
        x = 0;
        y = 1;
        return b;
    }
    int x1, y1;
    int d = extended_gcd(b % a, a, x1, y1);
    x = y1 - (b / a) * x1;
    y = x1;
    return d;
}

int main()
{
    int a, b, c;
    cin >> a >> b >> c;
    int x, y;
    int g = extended_gcd(a, b, x, y);
    if (c % g != 0)
    {
        cout << "Impossible" << endl;
    }
    else
    {
        cout << g << " " << x * (c / g) << " " << y * (c / g) << endl;
    }
}

File: cpp/foxford_2nd_year/9/2.py
def extended_gcd(a, b):
    if b == 0:
        return a, 1, 0
    gcd, x1, y1 = extended_gcd(b, a % b)
    x = y1
    y = x1 - (a // b) * y1
    return gcd, x, y

def solve(a, b, c):
    gcd, x, y = extended_gcd(a, b)
    
    # Проверка, существует ли решение
    if c % gcd != 0:
        print("Impossible")
    else:
        # Масштабируем коэффициенты
        x *= c // gcd
        y *= c // gcd
        print(gcd, x, y)

# Чтение данных
a, b, c = map(int, input().split())
solve(a, b, c)


